### ISSUE TITLE
Why constrain_points_to_ray function

### ISSUE THREAD

**Issue by Feng1909** [rmurai0610/MASt3R-SLAM#104]
Thank for your excellent work, MASt3R-SLAM achieves best reconstruction quality in our experiment. 

I have a question about some details. In the code, we implement `constrain_points_to_ray` to ensure that the point projection perfectly matches the pixel position when projected back to the image. But why use `F.normalize` for point-matching? In other words, after using `constrain_points_to_ray`, the result of `F.normalize` should be constant if we know K of the camera and we already *assume that each frame has a unique camera centre*?
